presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java

```diff
@@ -530,6 +551,13 @@ private synchronized void startSplit(PrioritizedSplitRunner split)
 
     private synchronized PrioritizedSplitRunner pollNextSplitWorker()
     {
+        if (memoryBasedSlowDownEnabled) {
+            MemoryUsage memoryUsage = MEMORY_MX_BEAN.getHeapMemoryUsage();
+            if (memoryUsage.getUsed() > memoryUsage.getCommitted() * memoryBasedSlowDownThreshold) {
```

Comment:
Can you jmh microbenchmark the cost of getting this bean calls -- I am concerned about introducing this in the hot path for each split.

Reply:
I don't think this is where you should plug in: This will simply allow running the existing splits in the waitingSplits -- all this is does is not release any more "scan/leaf" splits to mlsq -- it does nothing about intermediate splits eg. 

Did you consider teaching the MLSQ about this policy instead ? It will naturally stall the split runner threads.

Reply:
the point in time checks might not be representative of the overall health. If I am not wrong the memory allocation will follow a sawtooth pattern.
Relevant article from Elasticsearch guys where they rely on the fill rate of old gen to detect memory pressure.
https://www.elastic.co/blog/found-understanding-memory-pressure-indicator

Reply:
The memory usage is pulled out into a seperate thread so it won't impact the task/split processing.